src/main/scala/mesosphere/marathon/tasks/TaskTracker.scala

```diff
@@ -72,13 +72,19 @@ class TaskTracker @Inject() (
   def running(appId: PathId, status: TaskStatus): Future[MarathonTask] = {
     val taskId = status.getTaskId.getValue
     get(appId).find(_.getId == taskId) match {
-      case Some(stagedTask) =>
-        val task = stagedTask.toBuilder
+      case Some(oldTask) if !oldTask.hasStartedAt => // staged
+        val task = oldTask.toBuilder
           .setStartedAt(System.currentTimeMillis)
           .setStatus(status)
           .build
         getInternal(appId) += (task.getId -> task)
         store(appId, task).map(_ => task)
+
+      case Some(oldTask) if oldTask.hasStartedAt => // running
```

Comment:
This check is unnecessary, because it wouldn't get here if `hasStartedAt` is false.